### PR TITLE
Homework #3

### PR DESCRIPTION
**Pull request by AndryNosach**

### CODE REVIEW

File: Nosach/web-quiz-project/src/main/java/servlet/QuizDeleteServlet.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class QuizDeleteServlet extends HttpServlet {
+
+    private QuizService ts = QuizService.QUIZ_SERVICE;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
```
Comment (dinamio) [inline Nosach/web-quiz-project/src/main/java/servlet/QuizDeleteServlet.java:17]: Better to do deleting via delete method of HTTP

File: Nosach/web-quiz-project/src/main/java/servlet/QuizAddServlet.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import java.io.IOException;
+
+public class QuizAddServlet extends HttpServlet {
+
+    QuizService ts = QuizService.QUIZ_SERVICE;
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+        boolean formChecked = Boolean.parseBoolean(req.getParameter("formChecked"));
```
Comment (dinamio) [inline Nosach/web-quiz-project/src/main/java/servlet/QuizAddServlet.java]: Why do you need this formChecked parameter?

File: Nosach/web-quiz-project/src/main/webapp/view/list-of-quizzes.jsp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+            <c:forEach var="quizzes" items="${tests}">
+                <tr>
+                    <td>${quizzes.id}</td>
+                    <td>${quizzes.subject}</td>
+                    <td>${quizzes.theme}</td>
+                    <td>
+                        <a href="delete?id=${quizzes.id}" content="">Delete</a>
+                    </td>
+                    <td></td>
```
Comment (dinamio) [inline Nosach/web-quiz-project/src/main/webapp/view/list-of-quizzes.jsp]: Why do you need empty column?

File: Nosach/web-quiz-project/src/main/webapp/view/list-of-quizzes.jsp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                </tr>
+            </c:forEach>
+
+    </table>
+</div>
+<div name = "add">
+
+    <form name="add" action="add" method="post" >
```
Comment (dinamio) [inline Nosach/web-quiz-project/src/main/webapp/view/list-of-quizzes.jsp]: This can be href and get method, because we are just opening page

**Review by dinamio — APPROVED** [review]
Looks good. It's enough, to close homework, but I will not merge it, in case you want to do some changes, according to comments

File: Nosach/web-quiz-project/src/main/java/servlet/QuizDeleteServlet.java (hunk 1 above)
Comment (AndryNosach) [reply Nosach/web-quiz-project/src/main/java/servlet/QuizDeleteServlet.java:17]: didn't find how to send delete request via html href or form without js. Is it needed?

File: Nosach/web-quiz-project/src/main/java/servlet/QuizAddServlet.java (hunk 2 above)
Comment (AndryNosach) [reply Nosach/web-quiz-project/src/main/java/servlet/QuizAddServlet.java]: Request enters into AddServlet in two cases: 1. User clicked 'add' button on main  page, 2. User inserted data at "adding quiz page" into fields and clicked "submit".   formChecked parameter shows did we enter with 1st case ("false") or 2d ("true").

File: Nosach/web-quiz-project/src/main/webapp/view/list-of-quizzes.jsp (hunk 3 above)
Comment (AndryNosach) [reply Nosach/web-quiz-project/src/main/webapp/view/list-of-quizzes.jsp]: It's for adding "Modify Quiz" option in the future

File: Nosach/web-quiz-project/src/main/java/servlet/QuizDeleteServlet.java (hunk 1 above)
Comment (dinamio) [reply Nosach/web-quiz-project/src/main/java/servlet/QuizDeleteServlet.java:17]: You can do this with form <form method="delete"...>, or with js.
With html href you can't

File: Nosach/web-quiz-project/src/main/webapp/view/list-of-quizzes.jsp (hunk 3 above)
Comment (dinamio) [reply Nosach/web-quiz-project/src/main/webapp/view/list-of-quizzes.jsp]: Future will be in the future:)

File: Nosach/web-quiz-project/src/main/webapp/view/list-of-quizzes.jsp (hunk 4 above)
Comment (AndryNosach) [reply Nosach/web-quiz-project/src/main/webapp/view/list-of-quizzes.jsp]: Changed in last commit.

**Pull request merged**